### PR TITLE
Release/v2.8.2

### PR DESCRIPTION
**Pull request by lnxrp**
### PR Reminders

Be ware of the followings

- [ ] implement deployment script for your changes in deployment repo
- [ ] add verification in hardhat simulation(000-prepare-simulation-check, 902-newMarket-check or 903-simulation-check) and system test(901-system-test) in deployment repo
- [ ] update change log
 
---

1. **Contract**: make sure the code follows our convention, ex: naming, explicit returns, etc.; if uncertain, discuss with others

2. **Test**: make sure tests can cover most normal and edge cases; if not, open follow-up tickets. Also, look out for failed tests and fix them!

3. **CHANGELOG.md**: update when external interfaces are changed

4. Workflow: 
    - Github: assign the pr to yourself; if pairing, can merge directly; else, assign someone to help review
    - Asana: assign the corresponding ticket to yourself and leave the pr link on it for easier follow-ups
    - Discord: if someone is mentioned in this pr, tag on Discord

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205274587281447
  - https://app.asana.com/0/0/1205274587281459
  - https://app.asana.com/0/0/1205273975723510

### CODE REVIEW

File: contracts/MarketRegistry.sol [hunk 1]
```diff
@@ -130,11 +137,20 @@ contract MarketRegistry is IMarketRegistry, ClearingHouseCallee, MarketRegistryS
...
+        external
+        checkRatio(discountRatio)
+        onlyOwnerAndHottubFeeManager
+    {
         _feeDiscountRatioMap[trader] = discountRatio;
         emit FeeDiscountRatioChanged(trader, discountRatio);
     }
 
+    function setHottubFeeManager(address hottubFeeManagerArg) external onlyOwner {
+        _hottubFeeManager = hottubFeeManagerArg;
```
Comment (Wraecca) [inline contracts/MarketRegistry.sol]: strongly suggest to rename `_hottubFeeManager` to `_feeManager`. curie does not need to know anything about hot tub, it's just a feature enabling another role to update fee

this repo shouldn't include anything related to hot tub. hot tub is just a dapp built on top of curie

File: contracts/MarketRegistry.sol [hunk 2]
```diff
@@ -130,11 +137,20 @@ contract MarketRegistry is IMarketRegistry, ClearingHouseCallee, MarketRegistryS
         emit MarketMaxPriceSpreadRatioChanged(baseToken, ratio);
     }
 
-    function setFeeDiscountRatio(address trader, uint24 discountRatio) external checkRatio(discountRatio) onlyOwner {
+    function setFeeDiscountRatio(address trader, uint24 discountRatio)
+        external
+        checkRatio(discountRatio)
+        onlyOwnerAndHottubFeeManager
```
Comment (Wraecca) [inline contracts/MarketRegistry.sol]: why not onlyFeeManager but including owner to update feeDiscountRatio?

**Review by Wraecca — CHANGES_REQUESTED** [review]

**Comment by paco0x** [comment]
agree with wraecca, no other comments

File: contracts/MarketRegistry.sol [hunk 3]
```diff
@@ -130,11 +137,21 @@ contract MarketRegistry is IMarketRegistry, ClearingHouseCallee, MarketRegistryS
...
+        checkRatio(discountRatio)
+        onlyFeeManager
+    {
         _feeDiscountRatioMap[trader] = discountRatio;
         emit FeeDiscountRatioChanged(trader, discountRatio);
     }
 
+    function setFeeManager(address accountArg, bool isFeeManagerArg) external onlyOwner {
+        _feeManagerMap[accountArg] = isFeeManagerArg;
+        emit FeeManagerChanged(accountArg, isFeeManagerArg);
```
Comment (Wraecca) [inline contracts/MarketRegistry.sol:155]: minor feedback: now if we setFeeManager(alice, true) twice, it will emit FeeManagerChanged twice but actually FeeManager is not changed from the 2nd tx, suggest to do any of the following
1. revert if isFeeManagerArg == feeManagerMap[account]
2. only emit when it's diff
3. rename event to sth like....FeeManagerSet?

prefer 1, but if we think owner may want to set many fee manager at once in a batched tx then maybe 2 is easier (1 will break the batched tx easier if owner didn't check the state one by one)

Comment (lnxrp) [reply contracts/MarketRegistry.sol:155]: (2) seems to be better in your case because it is less intrusive (no-op > revert)

**Review by Wraecca — APPROVED** [review]
